env.go

```diff
@@ -128,6 +129,40 @@ func (v *boolValue) String() string {
 	return strconv.FormatBool(bool(*v))
 }
 
+type urlValue url.URL
+
+func newURLValue(x url.URL, p *url.URL) *urlValue {
+	*p = x
+	return (*urlValue)(p)
+}
+
+func (v *urlValue) Set(x string) error {
+	if x == "" {
+		return errors.New("empty")
+	}
+	u, err := url.Parse(x)
+	if err != nil {
+		return err
+	}
+	*v = urlValue(*u)
+	return err
+}
+
+func (v *urlValue) String() string {
+	u := url.URL{
```

Comment:
Why is this necessary? If it's to redact sensitive fields you could use v.Redacted() here instead.

Reply:
Fixed. For the record the reason I did this originally is that calling String on v recurses. I needed to coerce v back to a url.URL before I can call String.